### PR TITLE
add marko syntax highlighting

### PR DESCRIPTION
**Pull request by nyrosmith**
Associated Issue: #3160 

Here's the Pull Request Doc
https://devtools-html.github.io/debugger.html/CONTRIBUTING.html#pull-requests

### Test Plan
- [x] debug https://github.com/marko-js-samples/marko-webpack
- [x] open a .marko file
- [x] See if the source has syntax highlighting

### CODE REVIEW

**Comment by bomsy** [comment]
nice!

File: src/utils/source.js [hunk 1]
```diff
@@ -163,12 +163,18 @@ function getSourceLineCount(source: Source) {
...
 
   if (!text || isWasm) {
     return { name: "text" };
   }
 
+  // if the url ends with .marko we set the name to Javascript so
+  // syntax highlighting works for marko too
+  if (url.match(/\.marko$/i)) {
+    return { name: "javascript" };
+  }
```
Comment (jasonLaster) [inline src/utils/source.js]: mind adding a unit test to `utils/tests/source`

**Comment by jasonLaster** [comment]
looks like this affected the other unit tests

**Comment by jasonLaster** [comment]
could you share a screenshot? I'd love to share this on twitter

**Comment by jasonLaster** [comment]
added marko to debugger examples, it's a shame it's not hosted tho:
https://github.com/devtools-html/debugger-examples/commit/73c369084038af089d3b0cb77d7b948a163e9e27#diff-04c6e90faac2675aa89e2176d2eec7d8R42

**Comment by nyrosmith** [comment]
I'll attach a screenshot tomorrow and add a unit test.  But I could need help with the other test bc jest still shows lots of failing tests on my machine

**Comment by nyrosmith** [comment]
@jasonLaster here a gif you can share
marko

**Comment by nyrosmith** [comment]
Why is this still failing?

**Comment by codehag** [comment]
Its looking for identity equality, the issue is with jest/jasmine's toBe vs toEqual. toBe expects identity, toEqual expects same value

this can be fixed by changing
 
`expect(getMode(source)).toBe({ name: "javascript" });`

to 

`expect(getMode(source)).toEqual({ name: "javascript" });`

**Comment by nyrosmith** [comment]
Oh stupid me!

**Comment by codehag** [comment]
XD nah its all good. we all do it

**Comment by codehag** [comment]
It looks like `toBe` is still used in that test. the other test (prettyPrint) looks like an intermittent failure, so don't worry about that one :)

**Comment by codehag** [comment]
I'll take a look at the mochi test :)

**Comment by jasonLaster** [comment]
seeing two interesting failures:

6053 INFO TEST-UNEXPECTED-FAIL | devtools/client/debugger/new/test/mochitest/browser_dbg-breaking-from-console.js | Test timed out - 

6054 INFO TEST-UNEXPECTED-FAIL | devtools/client/debugger/new/test/mochitest/browser_dbg-breaking.js | Test timed out - 


it'd be nice to pull it down and run `yarn mochi browser_dbg-breaking`

**Comment by codehag** [comment]
I tried removing all of the changes, but one of the mochi tests is still failing. This is the reason: `Failed to retrieve MOZ_UPLOAD_DIR env var` -- i am stumped.. :/

**Comment by nyrosmith** [comment]
This is going to make me mad. Feel like mad hatter right now...

**Comment by codehag** [comment]
that makes two of us!

**Comment by wldcordeiro** [comment]
Going to try rebasing this on master and seeing how it goes. @nyrosmith

**Comment by wldcordeiro** [comment]
Rebased and ran mochitests with success and pushed the rebased branch. Let's see how CI goes. 👍

**Comment by wldcordeiro** [comment]
Looking good! Thanks @nyrosmith for the good work.

**Pull request merged**